Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Door.cs

```diff
@@ -0,0 +1,244 @@
+﻿// COPYRIGHT 2022 by the Open Rails project.
+// 
+// This file is part of Open Rails.
+// 
+// Open Rails is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// Open Rails is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Open Rails.  If not, see <http://www.gnu.org/licenses/>.
+using System.IO;
+using Orts.Common;
+using Orts.Parsers.Msts;
+using ORTS.Scripting.Api;
+
+namespace Orts.Simulation.RollingStocks.SubSystems
+{
+    // Door state: do not change the order of this enum
+    public enum DoorState
+    {
+        Closed,
+        Closing,
+        Open,
+        Opening,
```

Comment:
It would have been more logic to keep Opening between Closing and Open.
Opening is an intermediate state between Closed and Open.

Reply:
Yeah, that was my original intention. However, for the Train Driving Window @mbm-OR wanted to show "Doors opening" until all doors were open. The reasoning was that if a locomotive has no doors it would immediately report "Doors open" even if the rest of the doors in the train weren't open yet. Another solution could be to define the number of doors available in each vehicle, and exclude vehicles without doors.

Reply:
So do I revert this to Closed->Closing->Opening->Open for approval? I prefer this order too, but then some work has to be done for correct display at Train Driving Window

Reply:
Yes

Reply:
OK, done